addons/account/models/account_reconcile_model.py

```diff
@@ -115,6 +115,76 @@ def _validate_amount(self):
                 except re.error:
                     raise UserError(_('The regex is not valid'))
 
+    def _prepare_aml_vals(self, partner):
+        self.ensure_one()
+
+        taxes = self.tax_ids
+        if taxes and partner:
+            fiscal_position = self.env['account.fiscal.position']._get_fiscal_position(partner)
+            if fiscal_position:
+                taxes = fiscal_position.map_tax(taxes)
+
+        return {
+            'name': self.label,
+            'account_id': self.account_id.id,
+            'partner_id': partner.id,
+            'analytic_account_id': self.analytic_account_id.id,
+            'analytic_tag_ids': [Command.set(self.analytic_tag_ids.ids)],
+            'tax_ids': [Command.set(taxes.ids)],
+            'reconcile_model_id': self.model_id.id,
+            'journal_id': self.journal_id.id,
+        }
+
+    def _apply_line_for_manual_widget(self, residual_amount_currency, partner, currency):
+        self.ensure_one()
+
+        if self.amount_type == 'percentage':
+            amount_currency = currency.round(residual_amount_currency * (self.amount / 100.0))
+        elif self.amount_type == 'fixed':
+            sign = 1 if residual_amount_currency > 0.0 else -1
+            amount_currency = currency.round(self.amount * sign)
+
+        return {
+            **self._prepare_aml_vals(partner),
+            'currency_id': currency.id,
+            'amount_currency': amount_currency,
+        }
+
+    def _apply_line_for_bank_widget(self, residual_amount_currency, partner, st_line):
+        self.ensure_one()
+        currency = st_line.foreign_currency_id or st_line.journal_id.currency_id or st_line.company_currency_id
+
+        amount_currency = None
+        if self.amount_type == 'percentage_st_line':
+            amount_currency = currency.round(residual_amount_currency * (self.amount / 100.0))
+        elif self.amount_type == 'regex':
```

Comment:
percentage of balance/fixed not relevant, here?

edit: ok I see the call to `_apply_line_for_manual_widget()` but it's convoluted :dizzy_face: and not commented

Reply:
I don't like the idea to make such "gadgeto" method